internal/pkg/cli/app_deploy.go

```diff
@@ -328,6 +297,8 @@ func (o *appDeployOpts) configureClients() error {
 	// app deploy CF client against env account profile AND target environment region
 	o.appDeployCfClient = cloudformation.New(envSession)
 
+	o.s3Service = s3.New(envSession)
```

Comment:
I'm pretty sure these s3 buckets are in the project account - and will need to be written with the project session. (the buckets grant x-account read access, I believe)